types/react-autosuggest/index.d.ts

```diff
@@ -57,22 +58,11 @@ declare namespace Autosuggest {
     method: 'click' | 'enter';
   }
 
-  interface Theme {
-    container?: string;
-    containerOpen?: string;
-    input?: string;
-    inputOpen?: string;
-    inputFocused?: string;
-    suggestionsContainer?: string;
-    suggestionsContainerOpen?: string;
-    suggestionsList?: string;
-    suggestion?: string;
-    suggestionFirst?: string;
-    suggestionHighlighted?: string;
-    sectionContainer?: string;
-    sectionContainerFirst?: string;
-    sectionTitle?: string;
-  }
+  type ThemeKey = 'container' | 'containerOpen' | 'input' | 'inputOpen' | 'inputFocused' | 'suggestionsContainer' |
+    'suggestionsContainerOpen' | 'suggestionsList' | 'suggestion' | 'suggestionFirst' | 'suggestionHighlighted' |
+    'sectionContainer' | 'sectionContainerFirst' | 'sectionTitle';
+
+  type Theme = Partial<Record<ThemeKey, string | React.CSSProperties>>;
```

Comment:
I think the definition for `Theme` was incorrect.

As far as I can tell it should be closer to an `any`

The `ThemeKey` list of properties is just the default if you don't specify `theme`
https://github.com/moroshko/react-autosuggest#theme-optional

I'd change `Theme` to be...

```
type Theme = {
  [key: string]: string | React.CSSProperties;
} & Partial<Record<ThemeKey, string | React.CSSProperties>>;
```

Reply:
Is that correct? You *can* pass in extra properties, but react-autosuggest will ignore any keys not in ThemeKey.

Reply:
If you're using CSS Modules and you do this...

```js
const theme = require('./MyComponent.css');
```

...you're going to get an `any` as the result.

It would make this easier to use if the definition was the one I suggested rather then locked to specific props.